interface_notes.md

```diff
@@ -0,0 +1,387 @@
+pmw_tb.pmw_37pct           | list_numpy_to_numpy | Passive Microwave 37 MHz Polarization Corrected Temperature
+pmw_tb.pmw_89pct           | list_numpy_to_numpy |                                                            
+pmw_tb.pmw_color37         | list_numpy_to_numpy |                                                            
+pmw_tb.pmw_color89         | list_numpy_to_numpy |                                                            
+sfc_winds.windbarbs        | list_numpy_to_numpy |                                                            
+single_channel             | list_numpy_to_numpy |                                                            
+visir.Night_Vis            | list_numpy_to_numpy |                                                            
+visir.Night_Vis_GeoIPS1    | list_numpy_to_numpy |                                                            
+visir.Night_Vis_IR         | list_numpy_to_numpy |                                                            
+visir.Night_Vis_IR_GeoIPS1 | list_numpy_to_numpy |                                                            
+```
+
+# Important to do for review
+- ~~Ensure that the output of alg.get_list(by_type=True) fits what it was before. Something doesn't look right.~~
+  - I think I have this fixed. Still worth checking, but it doesn't look like this interface was working correctly anyway!
+
+# Questions
+- Need to figure out what we want to call `Boundaries` and `Gridlines` which are both yaml-config based. These likely
+  shouldn't be combined into a single class because they are sometimes mixed and matched.
+  - Maybe have a MapInterface module that defines which Boundaries and Gridlines you want to use.  
+- It looks like the `output_config` interface implements `get_output_format` and `get_output_format...`. These need to
+  be updated to work better with the new system and have appropriate names.
+- What is the difference between `geoips/dev/utils.py` and `geoips/geoips_utils.py`? It seems like a lot of what is in
+  dev should just be moved to the top-level module?
+- Should each of the interfaces have `geoips test <interface>` in the CLI?
+  - For example, for algorithms, this would call `algs.test_alg_interface()` and would look like `geoips test
+    algorithms`
+  - This would allow a user/developer to automatically test any new plugins that they've added to be sure that they're
+    working correctly with GeoIPS
+- What is `get_remove_duplicates_func()` in filename.py and title.py?
+- One advantage to using classes rather than modules for interface modules would be the ability to deprecate old
+  behavior. For example, currently, if I want to deprecate the use of `alg_func_type` in favor of `func_type` it becomes
+  fairly difficult. With a base class, this kind of thing would be relatively easy. I'm sure that there are some
+  significant disadvantages here, too, though. Just thinking.
+
+
+# Interfaces
```

Comment:
@jsolbrig I'm going to assume these are just comments you were making as you go - and they do not need to be updated for accuracy?